lib/net/http_xhr_plugin.js

```diff
@@ -62,10 +62,15 @@ shaka.net.HttpXHRPlugin = function(uri, request, requestType) {
             return all;
           },
           {});
-      var response = shaka.net.HttpPluginUtils.makeResponse(headers,
-          target.response, target.status, uri, target.responseURL,
-          requestType);
-      resolve(response);
+
+      try {
+        var response = shaka.net.HttpPluginUtils.makeResponse(headers,
+            target.response, target.status, uri, target.responseURL,
+            requestType);
+        resolve(response);
+      } catch (error) {
+        reject(error);
```

Comment:
If error is something other than shaka.util.Error, you'll want to wrap this.

Reply:
I don't know if this could be anything but a shaka.util.Error. This is catching thrown errors from makeResponse, which only throws an error in one place, and that's a shaka.util.Error. Nothing else in that function is something that could throw by itself, I don't think.

Reply:
Okay, fair enough.

Reply:
I actually agree with @joeyparrish, there should not be unchecked errors, so I added an assertion.